### ISSUE TITLE
Booth's algorithm

### ISSUE THREAD

**Issue by TobyLobster** [MarkusThill/65-C-02-Assembler#1]
Don't forget to initialize R+1 to zero at the start of this routine, otherwise it gives incorrect results.

**Comment by MarkusThill** [comment]
Thanks for the hint. I think R and R+1 should be initialized to zero in the last line (44) of the code, or not?

**Comment by TobyLobster** [comment]
Yes that does work if you only call the multiply routine once - but it will fail if the multiply routine is called a second time from within a larger program. The routine becomes more useful if you can call it multiple times.

**Comment by TobyLobster** [comment]
You may be interested in my comparison of multiply routines for the 6502, Booth's algorithm included: https://github.com/TobyLobster/multiply_test

**Comment by MarkusThill** [comment]
> Yes that does work if you only call the multiply routine once - but it will fail if the multiply routine is called a second time from within a larger program. The routine becomes more useful if you can call it multiple times.

Ahh, yes. I see. You are right. I will have to correct that...

**Comment by MarkusThill** [comment]
> You may be interested in my comparison of multiply routines for the 6502, Booth's algorithm included: https://github.com/TobyLobster/multiply_test

Wow! That is a really cool collection of algorithms. I will definetely have a closer look at it...
